website/src/blog/2020-03-18-entity-framework.md

```diff
@@ -0,0 +1,1050 @@
+
+But we can go further and even allow more. Lets say we want to allow the consumer to search for specific grades in our students enrolment list.
+
+In order to allow filtering on the enrollments we can add the same `UseFiltering` attribute in our entity on the `Enrollments` collection and this property becomes filterable.
+
+```csharp
+public class Student
+{
+    [Key]
+    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+    public int Id { get; set; }
+    public string LastName { get; set; }
+    public string FirstMidName { get; set; }
+    public DateTime EnrollmentDate { get; set; }
+
+    [UseFiltering]
+    public virtual ICollection<Enrollment> Enrollments { get; set; }
+}
+```
+
+We don\`t need to apply `UseSelections` again. `UseSelections` really only has to be applied where the data is fetched. In this case we do only want to support filtering but no sorting on enrollments. I could again add both but decided to only use filtering here.
+
+Let us restart our server and modify our query further.
+
+```bash
+dotnet run --urls http://localhost:5000
+```
+
+For the next query we will get all students with the last name `Bar` that are enrolled in the course with the `courseId` `1`.
+
+```graphql
+query {
+  students(where: { lastName: "Bar" }) {
+    firstMidName
+    lastName
+    enrollments(where: { courseId: 1 }) {
+      courseId
+      course {
+        title
+      }
+    }
+  }
+}
+```
+
+The following query translates again to a single SQL statement.
+
+```sql
+SELECT "s"."FirstMidName", "s"."LastName", "s"."Id", "t"."CourseId", "t"."Title", "t"."EnrollmentId", "t"."CourseId0"
+    FROM "Students" AS "s"
+    LEFT JOIN (
+        SELECT "e"."CourseId", "c"."Title", "e"."EnrollmentId", "c"."CourseId" AS "CourseId0", "e"."StudentId"
+        FROM "Enrollments" AS "e"
+        INNER JOIN "Courses" AS "c" ON "e"."CourseId" = "c"."CourseId"
+        WHERE "e"."CourseId" = 1
+    ) AS "t" ON "s"."Id" = "t"."StudentId"
+    WHERE "s"."LastName" = 'Bar'
+    ORDER BY "s"."Id", "t"."EnrollmentId", "t"."CourseId0"
+```
+
+With filtering and sorting we infer without almost no code complex filters from our code. This allows us to query our data with complex expressions while drilling into the data graph.
+
+_Hot Chocolate_ supports complex expressions with a variety of query operators that can be enabled by just adding a simple attribute on your field resolver. We can also configure the filter capabilities which we want to allow.
+
+## Paging
+
+But we still might get to much data back. What if we select all the students from a real university database. This is where our paging middleware comes in. The paging middleware implements the relay cursor pagination pattern.
+
+> Since we cannot do a skip while with entity framework we actually use a indexed based pagination underneath. For convenience we are wrapping this as really cursor pagination. With mongoDB and other database provider we are supporting real cursor base pagination.
```

Comment:
- 'an indexed' instead of 'a indexed'
- Did you mean 'relay' instead of 'really'?